spec/spec_models.cr

```diff
@@ -360,42 +375,64 @@ require "uuid"
     table_name uuids
 
     primary uuid : UUID, converter: Granite::Converters::Uuid(String), auto: false
-    field field_uuid : UUID, converter: Granite::Converters::Uuid(String)
+    field! field_uuid : UUID, converter: Granite::Converters::Uuid(String)
   end
 
   class TodoJsonOptions < Granite::Base
     adapter {{ adapter_literal }}
     table_name todos_json
 
-    field name : String, json_options: {key: "task_name"}
-    field priority : Int32, json_options: {ignore: true}
-    field updated_at : Time, json_options: {ignore: true}
-    field created_at : Time, json_options: {key: "posted"}
+    primary id : Int64
+
+    @[JSON::Field(key: "task_name")]
```

Comment:
yeah, I wouldn't want to see this in my codebase. 👎   I would prefer seeing a `src/serializers/TodoJsonOptionsSerializer.cr` that handles this.  The good news is this that this is optional.

Reply:
It's the easiest way.  One line annotation vs a whole new class seems a bit much. 

Is just the nature of `JSON::Serializable`.

Reply:
I vote for us to use annotations in the background while providing the user an unchangecahable dsl, I think is a good idea to move the validation and serialization into annotations as long as we do that  as a `implementation detail`, obscuring the user from it.

Reply:
This is easier said than done IMO.  I don't think you could get a similar syntax.

E.x. https://play.crystal-lang.org/#/r/78k2

Annotations have to be applied to the property when it's defined.

Reply:
I'm curious, how would you create a custom annotation if a validation depends on two fields like `password` and `password_confirmation`?

Reply:
Off the top of my head, pass the getter/ivar name as an annotation field.

```
class PasswordAssertion(ModelPropertyType)
  include CrSerializer::Assertions::Assertion

  @message : String = "'{{actual}}' is not a valid '{{field}}'"

  def initialize(
    field : String,
    message : String?,
    @actual : ModelPropertyType,
    @confirmation : String
  )
    super field, message
  end

  def valid? : Bool
	# Do something 
  end
end


@[PasswordAssertion(confirmation: password_confirmation)]
column password : String
```

Something like that maybe?

Reply:
Ok.  I can see that working.  I was trying to come up with a validation that is not `field/column` related or you would want it associated with multiple fields.